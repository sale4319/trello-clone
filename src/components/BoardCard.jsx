import React from 'react'
import { useHistory } from 'react-router';
import { Routes } from '../../constants/Routes';

export const BoardCard = ({ board, newBoard = false, handleNewBoardClick }) => {
    const history = useHistory();
    const handleBoardClick = () => {
        if (board) {
            history.push(`${Routes.Board}/${board.id}`);
        }
    };

    const boardName = board ? board.name : 'Create a new Board';
    return (
        <div
            className={`board-card-container ${newBoard ? 'board-card-color-new' : 'board-card-color-default'}`}
            onClick={newBoard ? handleNewBoardClick : handleBoardClick}>

            <div
                className={`board-card-text ${newBoard && 'board-card-text-center'}`}
                title={boardName}>
            </div>
        </div>
    );
};