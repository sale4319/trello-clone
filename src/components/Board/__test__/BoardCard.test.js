import React from 'react';
import { BoardCard } from '../BoardCard';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('component renders and takes snapshot', () => {
    const component = render(<BoardCard/>);
    expect(component.container).toMatchSnapshot();
  });