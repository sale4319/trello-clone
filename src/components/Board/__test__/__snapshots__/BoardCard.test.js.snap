// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders and takes snapshot 1`] = `
<div>
  <div
    class="board-card-container board-card-color-default"
  >
    <div
      class="board-card-text false"
      title="Create a new Board"
    >
      Create a new Board
    </div>
  </div>
</div>
`;
